app/controllers/users_controller.rb

```diff
@@ -0,0 +1,46 @@
+class UsersController < ApplicationController
+  before_action :set_user, only: [:show, :edit, :update, :destroy]
+  before_action :ensure_superuser
+
+  # GET /users
+  def index
+    @users = User.all
+  end
+
+  # GET /users/1
+  def show
+  end
+
+  # GET /users/1/edit
+  def edit
+  end
+
+  # PATCH/PUT /users/1
+  def update
+    if params[:toggle]
+      if @user.has_role? :admin
+        @user.remove_role :admin
+      else
+        @user.add_role :admin
+      end
+    end
+    redirect_to users_path
+  end
+
+  # DELETE /users/1
+  def destroy
+    @user.destroy
+    redirect_to users_url, notice: 'User was successfully destroyed.'
+  end
+
+  private
+    # Use callbacks to share common setup or constraints between actions.
+    def set_user
+      @user = User.find(params[:id])
+    end
+
+    # Only allow a trusted parameter "white list" through.
+    def user_params
+      params.fetch(:user, {})
```

Comment:
does it mean fetch all params? It will be safer to specify all of the params that we want to catch to avoid passing unwanted parameters.

e.g. adding `.permit(:id)`

also, idk whether it is better to use `.fetch(:user, {})` or `.require(:user)` because in a stackoverflow post, it is said that fetch will provide a default value if user is not defined, is it possible to happen? if it shouldn't happen i think that it will be better to use `require`.

Reply:
Actually, I did not use this method but I just leave it there in case one day I need it.

You can check at the update section, I am not using the :user at all, so it is actually possible. Thus, fetch is saver.

I also do not really know also why whitelisting is important, so in my update, I did not goes through the whitelisting at all. You can check how I used the params in my update, and let me know if you think that it is necessary for whitelisting that particular params.

Reply:
I see, it is usually used for specifying the parameters that you want to pass when creating/updating user, but if you don't use it, then I think it will be better to just remove it

Reply:
Ok. Removed.